.changeset/many-socks-warn.md

```diff
@@ -0,0 +1,9 @@
+---
+'@graphql-codegen/flow-operations': patch
+'@graphql-codegen/visitor-plugin-common': patch
+'@graphql-codegen/typescript-operations': patch
+---
+
+Add option `inlineFragmentTypes` for deep inlining fragment types within operation types. This `inlineFragmentTypes` is set to `true` by default.
```

Comment:
@dotansimha Thoughts on the option? I think if we would instead make it a string: `fragmentSpreadTypeMode` with values (`inline`, `fragment`), we can later add `mask` without introducing a new conflicting option

Reply:
I think `inlineFragmentTypes` or `inlineFragments` should be fine